rovercode/drivers/grovepi_chainable_rgb_leds.py

```diff
@@ -0,0 +1,50 @@
+"""Class for communicating with a GrovePi chainable RGB LED."""
+
+import os
+import logging
+import time
+
+logging.basicConfig()
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.getLevelName('INFO'))
+if os.getenv('DEVELOPMENT', 'false').lower() == 'true':
+    LOGGER.warning("In DEVELOPMENT mode. Using dummy.")
+    from drivers.dummy_grovepi_interface import chainableRgbLed_pattern, \
+        storeColor, chainableRgbLed_init, pinMode
+else:
+    from grovepi import chainableRgbLed_pattern, \
+        storeColor, chainableRgbLed_init, pinMode
+
+
+class GrovePiChainableRgbLeds:
+    """A module to set the GrovePi Chainable RGB LEDs."""
+
+    MODE_THIS_LED_ONLY = 0
+    COMPONENT_RANGE = range(0, 256)
+
+    port = None
+
+    def setup(self, port, count):
+        """Create a GrovePi Chainable RGB LEDs module."""
+        self.port = int(port)
+        pinMode(port, "OUTPUT")
+        time.sleep(1)
+        chainableRgbLed_init(self.port, count)
+        LOGGER.info("Setting up %s GrovePi Chainable RGB LEDs on port %s",
+                    count, port)
+
+    def set_color(self, led, red, green, blue):
+        """Set color in RGB."""
+        if self.port is None:
+            raise RuntimeError("Must call setup before setting color.")
+
+        for component in (red, green, blue):
+            if component not in self.COMPONENT_RANGE:
+                LOGGER.error(f'RGB color value %s not in range '
```

Comment:
I'm not sure how the other drivers are setup, but it might make more sense for a driver to raise an exception here (`ValueError`) and for something higher up to decide what to do about it (`LOGGER` right now). Maybe in the future, we'll want to do something more complicated that we won't want to put in the driver

Reply:
Good idea. I'll raise exceptions at the chainable_rgb_leds_manger.py level (since it's tested), and handle the exceptions with a log at the app.py level.